File: java/dsa/algorithms/sliding-window/min-size-subarray-sum.java
public class MinSizeSubarray {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int sum = 0;
        int minLen = Integer.MAX_VALUE;

        for (int right = 0; right < nums.length; right++) {
            // 1️⃣ Expand
            sum += nums[right];

            // 2️⃣ Shrink
            while (sum >= target) {
                // 3️⃣ Record
                if (right - left + 1 < minLen) {
                    minLen = right - left + 1;
                }

                // 4️⃣ Slide
                sum -= nums[left];
                left++;
            }
        }
        return minLen == Integer.MAX_VALUE ? 0 : minLen;
    }
}

File: python/dsa/algorithms/sliding-window/min-size-subarray-sum.py
def min_sub_array_len(target: int, nums: List[int]) -> int:
    left = 0
    total = 0
    min_len = float('inf')

    for right in range(len(nums)):
        # 1️⃣ Expand
        total += nums[right]

        # 2️⃣ Shrink
        while total >= target:
            # 3️⃣ Record
            if right - left + 1 < min_len:
                min_len = right - left + 1

            # 4️⃣ Slide
            total -= nums[left]
            left += 1

    return min_len if min_len != float('inf') else 0